internal/controller/postgrescluster/instance.go

```diff
@@ -1446,151 +1427,145 @@ func (r *Reconciler) reconcileUpgradeJob(
 
 	// TODO(tjmoore4): Checking the PG versions below, but in the future, version
 	// validation should be added as a webhook.
-	if cluster.Spec.Upgrade != nil && *cluster.Spec.Upgrade.Enabled {
-		if cluster.Spec.PostgresVersion > cluster.Spec.Upgrade.FromPostgresVersion {
-			// before creating the upgrade job, observe all resources currently
-			// relevant to reconciling data sources, and update status accordingly
-			endpoints, upgradeJob, err := r.observeUpgradeEnv(ctx, cluster)
-			if err != nil {
-				return false, errors.WithStack(err)
-			}
+	// before creating the upgrade job, observe all resources currently
+	// relevant to reconciling data sources, and update status accordingly
+	endpoints, existingUpgradeJob, err := r.observeUpgradeEnv(ctx, cluster)
+	if err != nil {
+		return false, errors.WithStack(err)
+	}
 
-			if upgradeJob == nil {
-				err = r.prepareForUpgrade(ctx, cluster, observed, endpoints, upgradeJob)
-				if err != nil {
-					return false, errors.WithStack(err)
-				}
+	if cluster.Spec.Upgrade == nil || !*cluster.Spec.Upgrade.Enabled {
```

Comment:
My thought here is that we still would want to update the PostgresCluster `status` and `conditions` (if necessary) even if upgrading is disabled.  For instance, if someone disables upgrading while an upgrade is in progress, we would still want to update the `status` and `conditions` appropriately once the upgrade Job completes.

And since `status` and `conditions` are set in `observeUpgradeEnv()`, I decided to leave that call in front of checking whether upgrading is enabled to ensure there is always a chance to for a status update (if one is needed).